src/khoj/routers/api_config.py

```diff
@@ -0,0 +1,359 @@
+@requires(["authenticated"])
+async def set_config_data(
+    request: Request,
+    updated_config: FullConfig,
+    client: Optional[str] = None,
+):
+    user = request.user.object
+    await map_config_to_db(updated_config, user)
+
+    configuration_update_metadata = {}
+
+    enabled_content = await sync_to_async(EntryAdapters.get_unique_file_types)(user)
+
+    if state.config.content_type is not None:
+        configuration_update_metadata["github"] = "github" in enabled_content
+        configuration_update_metadata["notion"] = "notion" in enabled_content
+        configuration_update_metadata["org"] = "org" in enabled_content
+        configuration_update_metadata["pdf"] = "pdf" in enabled_content
+        configuration_update_metadata["markdown"] = "markdown" in enabled_content
+
+    if state.config.processor is not None:
+        configuration_update_metadata["conversation_processor"] = state.config.processor.conversation is not None
+
+    update_telemetry_state(
+        request=request,
+        telemetry_type="api",
+        api="set_config",
+        client=client,
+        metadata=configuration_update_metadata,
+    )
+    return state.config
+
+
+@api_config.post("/data/content-source/github", status_code=200)
+@requires(["authenticated"])
+async def set_content_config_github_data(
+    request: Request,
+    updated_config: Union[GithubContentConfig, None],
+    client: Optional[str] = None,
+):
+    _initialize_config()
+
+    user = request.user.object
+
+    try:
+        await adapters.set_user_github_config(
+            user=user,
+            pat_token=updated_config.pat_token,
+            repos=updated_config.repos,
+        )
+    except Exception as e:
+        logger.error(e, exc_info=True)
+        raise HTTPException(status_code=500, detail="Failed to set Github config")
+
+    update_telemetry_state(
+        request=request,
+        telemetry_type="api",
+        api="set_content_config",
+        client=client,
+        metadata={"content_type": "github"},
+    )
+
+    return {"status": "ok"}
+
+
+@api_config.post("/data/content-source/notion", status_code=200)
+@requires(["authenticated"])
+async def set_content_config_notion_data(
+    request: Request,
+    updated_config: Union[NotionContentConfig, None],
+    client: Optional[str] = None,
+):
+    _initialize_config()
+
+    user = request.user.object
+
+    try:
+        await adapters.set_notion_config(
+            user=user,
+            token=updated_config.token,
+        )
+    except Exception as e:
+        logger.error(e, exc_info=True)
+        raise HTTPException(status_code=500, detail="Failed to set Github config")
+
+    update_telemetry_state(
+        request=request,
+        telemetry_type="api",
+        api="set_content_config",
+        client=client,
+        metadata={"content_type": "notion"},
+    )
+
+    return {"status": "ok"}
+
+
+@api_config.delete("/data/content-source/{content_source}", status_code=200)
```

Comment:
This also isn't being used anymore.

Reply:
This is being used in the web interface to delete data for each content-source (desktop, github or notion)